### PR TITLE
bgpv1: remove references to advertisement from CiliumBGPPeeringPolicy

### PR DESCRIPTION
**Pull request by harsimran-pabla**
Advertisement field got introduced into CiliumBGPFamily type when adding v2 APIs. This field is only required in new CiliumBGPPeerConfig structures. This change removes advertisement from CiliumBGPFamily and introduces new type CiliumBGPFamilyWithAdverts which will be used in v2 APIs.

There is no impact to current usage of the CiliumBGPPeeringPolicy CRD.

### CODE REVIEW

**Review by rastislavs — APPROVED** [review]
LGTM

**Comment by harsimran-pabla** [comment]
Ginkgo test failure ""Policy map sync fixed errors, consider running with debug verbose = policy to get detailed dumps""  : https://github.com/cilium/cilium/issues/29727

**Review by rastislavs — APPROVED** [review]

File: pkg/k8s/apis/cilium.io/register.go [hunk 1]
```diff
@@ -15,5 +15,5 @@ const (
 	//
 	// Maintainers: Run ./Documentation/check-crd-compat-table.sh for each release
 	// Developers: Bump patch for each change in the CRD schema.
-	CustomResourceDefinitionSchemaVersion = "1.29.0"
+	CustomResourceDefinitionSchemaVersion = "1.29.1"
```
Comment (danehans) [inline pkg/k8s/apis/cilium.io/register.go:18]: Why does this change require a CRD schema version bump?

**Comment by danehans** [comment]
> Advertisement field got introduced into CiliumBGPFamily type when adding v2 APIs. This field is only required in new CiliumBGPPeerConfig structures.

The field is a pointer and includes the `omitempty` json stuct tag so the field is not required. Can you clarify?

File: pkg/k8s/apis/cilium.io/v2alpha1/bgp_peer_types.go [hunk 2]
```diff
@@ -105,6 +105,11 @@ type CiliumBGPFamily struct {
 	// +kubebuilder:validation:Enum=unicast;multicast;mpls_label;encapsulation;vpls;evpn;ls;sr_policy;mup;mpls_vpn;mpls_vpn_multicast;route_target_constraints;flowspec_unicast;flowspec_vpn;key_value
 	// +kubebuilder:validation:Required
 	Safi string `json:"safi"`
+}
+
+// CiliumBGPFamilyWithAdverts represents a AFI/SAFI address family pair along with reference to BGP Advertisements.
+type CiliumBGPFamilyWithAdverts struct {
+	CiliumBGPFamily `json:",inline"`
```
Comment (danehans) [inline pkg/k8s/apis/cilium.io/v2alpha1/bgp_peer_types.go:112]: Why is CiliumBGPFamilyWithAdverts needed when it has the same fields as CiliumBGPFamily?

Comment (harsimran-pabla) [reply pkg/k8s/apis/cilium.io/v2alpha1/bgp_peer_types.go:112]: CiliumBGPFamily will not contain Advertisements after this change.

**Comment by harsimran-pabla** [comment]
Hi @danehans 

> The field is a pointer and includes the `omitempty` json stuct tag so the field is not required. Can you clarify?

I mean functionality of Advertisements is only required in BGPv2 CiliumBGPPeerConfig resource. It is indeed an optional field in that context. However, it should not be present at all in CiliumBGPPeeringPolicy which have references to CiliumBGPFamily.

File: pkg/k8s/apis/cilium.io/register.go (hunk 1 above)
Comment (harsimran-pabla) [reply pkg/k8s/apis/cilium.io/register.go:18]: We are modifying the schema - removing field in CiliumBGPFamily. I believe on upgrade if schema version is not modified, CRDs will not be updated. ( I am going by the comment at line 17 )

**Review by danehans — APPROVED** [review]
Thanks for providing clarification for the changes.

**Review by christarazi — APPROVED** [review]

**Comment by aanm** [comment]
@harsimran-pabla I assume this PR was backported in https://github.com/cilium/cilium/pull/30531? If yes, I've removed the needs-backport/1.15 label

**Pull request merged**